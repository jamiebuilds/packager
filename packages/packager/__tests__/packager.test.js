'use strict';
// @flow
const packager = require('../');
const path = require('path');

const fixtures = path.join(__dirname, '..', '__fixtures__');

let fixture = (name) => path.join(fixtures, name, 'entry.js');

test('sync', async () => {
  expect(await packager(fixture('sync'))).toMatchSnapshot();
});

test('async', async () => {
  expect(await packager(fixture('async'))).toMatchSnapshot();
});
