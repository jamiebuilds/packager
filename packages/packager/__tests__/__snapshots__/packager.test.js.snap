// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async 1`] = `
Object {
  "edges": Object {
    "/../packages/packager/__fixtures__/async/entry.js": Array [
      Object {
        "async": true,
        "filePath": "/../packages/packager/__fixtures__/async/moduleA.js",
      },
      Object {
        "async": true,
        "filePath": "/../packages/packager/__fixtures__/async/moduleB.js",
      },
    ],
  },
  "nodes": Object {
    "/../packages/packager/__fixtures__/async/entry.js": Object {
      "fileContents": "import(\\"./moduleA\\");
import(\\"./moduleB\\");
",
      "filePath": "/../packages/packager/__fixtures__/async/entry.js",
    },
    "/../packages/packager/__fixtures__/async/moduleA.js": Object {
      "fileContents": "console.log('moduleA');
",
      "filePath": "/../packages/packager/__fixtures__/async/moduleA.js",
    },
    "/../packages/packager/__fixtures__/async/moduleB.js": Object {
      "fileContents": "console.log('moduleB');
",
      "filePath": "/../packages/packager/__fixtures__/async/moduleB.js",
    },
  },
}
`;

exports[`sync 1`] = `
Object {
  "edges": Object {
    "/../packages/packager/__fixtures__/sync/entry.js": Array [
      Object {
        "async": false,
        "filePath": "/../packages/packager/__fixtures__/sync/moduleA.js",
      },
      Object {
        "async": false,
        "filePath": "/../packages/packager/__fixtures__/sync/moduleB.js",
      },
    ],
  },
  "nodes": Object {
    "/../packages/packager/__fixtures__/sync/entry.js": Object {
      "fileContents": "import \\"./moduleA\\";
import \\"./moduleB\\";
",
      "filePath": "/../packages/packager/__fixtures__/sync/entry.js",
    },
    "/../packages/packager/__fixtures__/sync/moduleA.js": Object {
      "fileContents": "console.log('moduleA');
",
      "filePath": "/../packages/packager/__fixtures__/sync/moduleA.js",
    },
    "/../packages/packager/__fixtures__/sync/moduleB.js": Object {
      "fileContents": "console.log('moduleB');
",
      "filePath": "/../packages/packager/__fixtures__/sync/moduleB.js",
    },
  },
}
`;
